### PR TITLE
1. add dropEvent

### PR DESCRIPTION
**Pull request by BeyondXinXin**

### CODE REVIEW

**Comment by BeyondXinXin** [comment]
Came across this great project，This interaction is the best I've ever met.
Thanks for sharing.

When I drag "*.Alz" to the main screen, I find it can not be opened.

**Comment by juzzlin** [comment]
Cool! I'll check and test this soon :)

File: src/application.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// This file is part of Heimer.
+﻿// This file is part of Heimer.
```
Comment (juzzlin) [inline src/application.cpp:1]: Not sure what's changed here? BOM?

File: src/editor_view.hpp [hunk 2]
```diff
@@ -174,6 +181,8 @@ public slots:
     MainContextMenu * m_mainContextMenu;
 
     bool m_gridVisible = false;
+
+    QString m_dropFile{};
```
Comment (juzzlin) [inline src/editor_view.hpp:185]: {} not needed

File: src/editor_view.hpp [hunk 3]
```diff
@@ -83,12 +85,17 @@ public slots:
 
     void wheelEvent(QWheelEvent * event) override;
 
+    void dropEvent(QDropEvent *event) override;
```
Comment (juzzlin) [inline src/editor_view.hpp:88]: Placement of '*' is incorrect. Please use `src/.clang-format`.

**Review by juzzlin — CHANGES_REQUESTED** [review]
Seems to work, but I noticed one bigger issue: if we have a modified mind map and we drag'n'drop a new one Heimer doesn't say that the current mind map is changed - it just replaces the active mind map with the new one. I guess that this should work like opening a new mind map in this case.

**Comment by BeyondXinXin** [comment]
I'm still a beginner, thanks for pointing out the mistakes.

**Comment by juzzlin** [comment]
Sorry for being a bit unresponsive, but I've been super busy this week. I still want to have this merged, so no worries :)

**Comment by juzzlin** [comment]
I squashed your commits into a single commit and pushed to master now. This was very good work! Thank you very much! :)

**Pull request closed without merging**